### PR TITLE
Add option to avoid dumping AUTO_INCREMENT values

### PR DESCRIPTION
**Pull request by mpdude**
We're using `SHOW CREATE TABLE` to get the table creation SQL, but that includes the `AUTO_INCREMENT` value (at least for some storage engines).

Depending on the use case, it would be nice to be able to get rid of that.

### CODE REVIEW

File: src/Webfactory/Slimdump/Database/Dumper.php [hunk 1]
```diff
@@ -37,12 +37,20 @@ public function enableForeignKeys()
...
         $this->output->writeln("DROP TABLE IF EXISTS `$table`;");
 
+        $tableCreationCommand = $db->fetchColumn("SHOW CREATE TABLE `$table`", array(), 1);
+
+        if ($noAutoIncrement) {
+            $tableCreationCommand = preg_replace('/ AUTO_INCREMENT=[0-9]*/', '', $tableCreationCommand);
+        }
+
+        $this->output->writeln($tableCreationCommand.";\n", OutputInterface::OUTPUT_RAW);
         $this->output->writeln($db->fetchColumn("SHOW CREATE TABLE `$table`", array(), 1).";\n");
```
Comment (mpdude) [inline src/Webfactory/Slimdump/Database/Dumper.php]: @Dword123 at least this line needs to be removed

**Comment by mpdude** [comment]
@Dword123 this is as we discussed, but now that I see it I find it somewhat ugly how the `$noAutoIncrement` flag needs to be passed along into the `dump()` method. 

Would it make things nicer if it were not a command flag but instead a configuration setting in the XML config?

/cc @dritter

**Comment by Dword123** [comment]
@mpdude Makes sense. Is this a new branch/pull request then?

**Comment by mpdude** [comment]
No, I'd change it on this branch and we can eventually squash the changes.

**Comment by Dword123** [comment]
How about that? It is now controlled by a config attribute.

File: src/Webfactory/Slimdump/SlimdumpCommand.php [hunk 2]
```diff
@@ -35,17 +35,21 @@ protected function configure()
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         $dsn = $input->getArgument('dsn');
+        $noAutoIncrement = $input->getOption('no-auto-increment');
```
Comment (mpdude) [inline src/Webfactory/Slimdump/SlimdumpCommand.php]: @Dword123 can this be removed?

File: src/Webfactory/Slimdump/SlimdumpCommand.php [hunk 3]
```diff
@@ -35,17 +35,21 @@ protected function configure()
...
 
     /**
      * @param Config $config
      * @param Connection $db
+     * @param OutputInterface $output
+     * @param boolean $noAutoIncrement
+     * @throws \Doctrine\DBAL\DBALException
      */
-    public function dump(Config $config, Connection $db, OutputInterface $output)
+    public function dump(Config $config, Connection $db, OutputInterface $output, $noAutoIncrement)
```
Comment (mpdude) [inline src/Webfactory/Slimdump/SlimdumpCommand.php]: @Dword123  Do you really use this parameter?

File: src/Webfactory/Slimdump/Config/Table.php [hunk 4]
```diff
@@ -72,6 +90,14 @@ public function isDataDumpRequired()
     }
 
     /**
+     * @return boolean
+     */
+    public function isAutoIncrement()
```
Comment (mpdude) [inline src/Webfactory/Slimdump/Config/Table.php]: @Dword123 This method name and the name of the setting is not so good IMO. What about something like "keep-auto-increment" or "keep-auto-increment-id" or the like?

File: src/Webfactory/Slimdump/SlimdumpCommand.php (hunk 2 above)
Comment (Dword123) [inline src/Webfactory/Slimdump/SlimdumpCommand.php]: @mpdude Yes it can be removed.

File: src/Webfactory/Slimdump/SlimdumpCommand.php (hunk 3 above)
Comment (Dword123) [inline src/Webfactory/Slimdump/SlimdumpCommand.php]: It's used to query the auto-increment parameter from the table config. Any other suggestion? Passing the boolean directly?

File: src/Webfactory/Slimdump/Config/Table.php (hunk 4 above)
Comment (Dword123) [inline src/Webfactory/Slimdump/Config/Table.php]: @mpdude keep-auto-increment sounds nice.

File: src/Webfactory/Slimdump/SlimdumpCommand.php (hunk 3 above)
Comment (mpdude) [inline src/Webfactory/Slimdump/SlimdumpCommand.php]: I mean, do you really use it when you update the `execute()` method as discussed above?

**Comment by mpdude** [comment]
Nice! Now could you please mention this feature in the README and also squash the commits?

**Comment by Dword123** [comment]
@mpdude Updated everything. I converted the parameter in dumpSchema() to just the boolean attribute that is needed in this place.

How do I squash the commits?

**Pull request merged**